central/processbaseline/datastore/datastore_impl.go

```diff
@@ -399,15 +399,15 @@ func (ds *datastoreImpl) CreateUnlockedProcessBaseline(ctx context.Context, key
 		return nil, err
 	}
 
-	// Build the elements for the baseline
-	elements := make([]*storage.BaselineItem, 0, len(baselineList))
+	// Build the de-duped elements for the baseline
+	elements := make(map[string]*storage.BaselineItem, len(baselineList))
 
 	for _, indicator := range baselineList {
 		baselineItem := processBaselinePkg.BaselineItemFromProcess(indicator)
 
 		insertableElement := &storage.BaselineItem{Item: &storage.BaselineItem_ProcessName{ProcessName: baselineItem}}
 
-		elements = append(elements, insertableElement)
+		elements[baselineItem] = insertableElement
```

Comment:
Wdyt about sorting alphabetically to provide a consistent output

Reply:
@connorgorman I guess that depends.  Looks like the UI is sorting them when they are presented, so not a lot of value in doing it for our UI customers.  they weren't previously sorted in the backend so they would be in whatever order they arrived.  We could sort them, but we would probably want to do that right before calling Upsert.  I'm not certain how valuable that is or would be as part of this fix.  Up to you.

Reply:
Works for me